shotgun_api3/shotgun.py

```diff
@@ -2219,33 +2215,16 @@ def _upload_to_storage(self, entity_type, entity_id, path, field_name, display_n
 
         # Step 1: get the upload url
 
-        upload_info = self._get_attachment_upload_info(is_thumbnail, filename)
+        multipart_upload = (os.path.getsize(path)  > 20000000)
 
-        # Step 2: upload the file
+        upload_info = self._get_attachment_upload_info(is_thumbnail, filename, multipart_upload)
 
-        fd = open(path, "rb")
-        try:
-            content_type = mimetypes.guess_type(filename)[0]
-            content_type = content_type or "application/octet-stream"
-            file_size = os.fstat(fd.fileno())[stat.ST_SIZE]
-
-            # Perform the request
-            opener = urllib2.build_opener(urllib2.HTTPHandler)
-
-            request = urllib2.Request(upload_info["upload_url"], data=fd)
-            request.add_header("Content-Type", content_type)
-            request.add_header("Content-Length", file_size)
-            request.get_method = lambda: "PUT"
-            result = opener.open(request)
-        except urllib2.HTTPError, e:
-            if e.code == 500:
-                raise ShotgunError("Server encountered an internal error.\n%s\n%s\n\n" % (url, e))
-            else:
-                raise ShotgunError("Unanticipated error occurred uploading %s: %s" % (path, e))
-        finally:
-            fd.close()
+        # Step 2: upload the file
 
-        LOG.debug("File uploaded to Cloud storage: %s", filename)
+        if multipart_upload:
+            self._multipart_upload_file_to_storage(path, upload_info)
```

Comment:
minor: some comments explaining why we need to make the distinction here based on file size could helpful.